python/lsst/summit/utils/guiders/transformation.py

```diff
@@ -0,0 +1,714 @@
+    irot[1] = rot[1].transpose()
+    irot[2] = rot[2].transpose()
+    irot[3] = rot[3].transpose()
+
+    # number of 90deg CCW rotations
+    nq = np.mod(det_nquarter, 4)
+
+    # get LL,Size of the CCD view BBox
+    llpt_ccd = Extent2D(bbox_ccd.getCorners()[0])
+    nx, ny = bbox_ccd.getDimensions()
+
+    # get translations to use for each NQ value
+    boxtranslation = {}
+    boxtranslation[0] = Extent2D(0, 0)
+    boxtranslation[1] = Extent2D(ny - 1, 0)
+    boxtranslation[2] = Extent2D(nx - 1, ny - 1)
+    boxtranslation[3] = Extent2D(0, nx - 1)
+
+    # build the forware Transform
+    ftranslation = AffineTransform.makeTranslation(-llpt_ccd)
+    frotation = AffineTransform(rot[nq])
+    fboxtranslation = AffineTransform.makeTranslation(boxtranslation[nq])
+    # ordering is third*second*first
+    forwards = fboxtranslation * frotation * ftranslation
+
+    btranslation = AffineTransform.makeTranslation(llpt_ccd)
+    brotation = AffineTransform(irot[nq])
+    bboxtranslation = AffineTransform.makeTranslation(-boxtranslation[nq])
+    backwards = btranslation * brotation * bboxtranslation
+
+    return forwards, backwards
+
+
+def mk_roi_bboxes(md, camera):
+    """
+    Make bounding box for a Guider stamp in the full CCD view
+
+    Parameters
+    ----------
+    md : lsst.daf.base.PropertyList
+        Metadata from one Guider CCD
+
+    camera: lsst.obs.lsst.LsstCam
+        LsstCam object
+
+    Returns
+    -------
+    ccd_view_bbox : Box2I
+        Bounding box for the stamps inside the full CCD in CCD view
+    """
+
+    # get need info from the stamp metadata
+    roiCol = md["ROICOL"]
+    roiRow = md["ROIROW"]
+    roiCols = md["ROICOLS"]
+    roiRows = md["ROIROWS"]
+
+    # get detector and amp
+    detector, ampName = get_detector_amp(md, camera)
+    amp = detector[ampName]
+
+    # get corner0 of the location for the ROI
+    lct = LsstCameraTransforms(camera, detector.getName())
+    corner0_CCDX, corner0_CCDY = lct.ampPixelToCcdPixel(roiCol, roiRow, ampName)
+    corner0_CCDX = int(corner0_CCDX)
+    corner0_CCDY = int(corner0_CCDY)
+
+    # get opposite corner, corner2, of the ROI
+    # depending on the RawFlipX,Y it could be either below or above corner0
+    if amp.getRawFlipX():
+        corner2_CCDX = corner0_CCDX - (roiCols - 1)
+    else:
+        corner2_CCDX = corner0_CCDX + (roiCols - 1)
+
+    if amp.getRawFlipY():
+        corner2_CCDY = corner0_CCDY - (roiRows - 1)
+    else:
+        corner2_CCDY = corner0_CCDY + (roiRows - 1)
+
+    # now make the CCD view BBox, here we want the LowerLeft Point to be the
+    # smallest x,y
+    ll_x = min(corner0_CCDX, corner2_CCDX)
+    ur_x = max(corner0_CCDX, corner2_CCDX)
+    ll_y = min(corner0_CCDY, corner2_CCDY)
+    ur_y = max(corner0_CCDY, corner2_CCDY)
+    ll_CCD = Point2D(ll_x, ll_y)
+    ur_CCD = Point2D(ur_x, ur_y)
+    ccd_view_bbox = Box2I(Box2D(ll_CCD, ur_CCD))
+
+    return ccd_view_bbox
+
+
+def get_detector_amp(md, camera):
```

Comment:
missing mypy annotations, including return type

Reply:
Do you know where can I get the metadata type?